setup_py_upgrade.py

```diff
@@ -134,11 +134,23 @@ def _list_as_str(lst: Sequence[str]) -> str:
         return '\n' + '\n'.join(lst)
 
 
-def reformat_lists(section: Dict[str, Any]) -> Dict[str, Any]:
-    return {
-        k: _list_as_str(v) if isinstance(v, (list, tuple)) else v
-        for k, v in section.items()
-    }
+def _dict_as_str(dct: Dict[str, str]) -> str:
+    return '\n' + '\n'.join(
+        (key + ' = ' + value) for key, value in dct.items()
+    )
+
+
+def reformat_values(section: Dict[str, Any]) -> Dict[str, Any]:
+    new_section = {}
+    for key, value in section.items():
+        if isinstance(value, (list, tuple)):
+            new_value = _list_as_str(value)
+        elif isinstance(value, dict):
+            new_value = _dict_as_str(value)
+        else:
+            new_value = value
```

Comment:
it'd be slightly shorter to just `new_section[key] = ...` for each of the three branches (less state to think about as well)